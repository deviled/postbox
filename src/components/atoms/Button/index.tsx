import styled, { css } from 'styled-components';
import {
  palette, PaletteProps, spacing, SpacingProps,
} from '@material-ui/system';

type ButtonProps = PaletteProps & SpacingProps;

export const Button = styled.button<ButtonProps>(({ theme }) => css<ButtonProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: ${theme.spacing(16, 32)};
  cursor: pointer;
  appearance: none;
  text-decoration: none;
  color: ${theme.palette.white};
  font-size: ${theme.typography.fontSize[16]};
  border: none;
  background-color: ${theme.palette.primary};

  &:hover {
    filter: brightness(85%);
  }

  &:disabled {
    filter: initial;
    background-color: ${theme.palette.background};
  }
  
  ${palette};
  ${spacing};
`);
