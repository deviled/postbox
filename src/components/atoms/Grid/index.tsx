import styled from 'styled-components';
import {
  flexbox, FlexboxProps, palette, PaletteProps, sizing, SizingProps, spacing, SpacingProps,
} from '@material-ui/system';

export type FlexProps = FlexboxProps & SpacingProps & PaletteProps & SizingProps;

export const FlexBox = styled.div<FlexProps>`
  display: flex;
  ${sizing}
  ${palette};
  ${spacing};
  ${flexbox};
`;

export const FlexCol = styled(FlexBox)`
  flex-direction: column; 
`;
