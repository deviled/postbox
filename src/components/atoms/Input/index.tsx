import styled, { css } from 'styled-components';
import {
  border, BordersProps, palette, PaletteProps, sizing, SizingProps, spacing, SpacingProps,
} from '@material-ui/system';

type InputProps = SpacingProps & PaletteProps & BordersProps & SizingProps;

export const Input = styled.input<InputProps>(({ theme }) => css<InputProps>`
  font-size: ${theme.typography.fontSize[16]};
  color: ${theme.palette.text};
  padding: ${theme.spacing(12)};
  border: 1px solid ${theme.palette.divider};
  background-color: ${theme.palette.white};

  ${spacing};
  ${palette};
  ${border};
  ${sizing};
  
  &:hover {
    border-color: ${theme.palette.text};
  }
`);
