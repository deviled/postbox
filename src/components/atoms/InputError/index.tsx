import React, { FC } from 'react';
import styled, { css } from 'styled-components';

const StyledError = styled.div(({ theme }) => css`
  font-size: ${theme.fontSize[12]};
  color: ${theme.palette.error};
`);

export const InputError: FC = ({ children }) => (
  <StyledError>{children}</StyledError>
);
