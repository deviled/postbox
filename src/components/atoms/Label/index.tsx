import React, { ComponentPropsWithoutRef, FC } from 'react';
import styled, { css } from 'styled-components';

type ILabelProps = ComponentPropsWithoutRef<'label'>;

const StyledLabel = styled.label(({ theme }) => css`
  display: flex;
  flex-direction: column;
  margin-bottom: ${theme.spacing(16)};
  
  &:last-of-type {
    margin-bottom: ${theme.spacing(24)};
  }
`);

export const Label: FC<ILabelProps> = ({ children, ...rest }) => (
  <StyledLabel {...rest}>{children}</StyledLabel>
);
