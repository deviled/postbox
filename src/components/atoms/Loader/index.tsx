import React, { FC } from 'react';

export const Loader: FC = () => <span>Loading ...</span>;
