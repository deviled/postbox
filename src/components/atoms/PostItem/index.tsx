import React, { FC } from 'react';
import styled, { css } from 'styled-components';
import { formatDate } from 'tools/formatDate';
import { IPost } from 'types';

const Article = styled.article(({ theme }) => css`
  font-size: ${theme.fontSize[16]};
  padding: ${theme.spacing(4, 16)};
  border: ${theme.borders[1]};
  margin-bottom: ${theme.spacing(16)};
`);

interface IPostItemProps {
  createdTime: IPost['created_time'];
}

export const PostItem: FC<IPostItemProps> = ({ children, createdTime }) => (
  <Article>
    <p>{formatDate(createdTime)}</p>
    <hr />
    <p>{children}</p>
  </Article>
);
