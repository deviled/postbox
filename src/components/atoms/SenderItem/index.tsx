import React, { ComponentPropsWithRef, forwardRef } from 'react';
import { ISender } from 'types';
import styled, { css } from 'styled-components';

interface IStyledButtonProps {
  isActive?: boolean;
}

const StyledButton = styled.button<IStyledButtonProps>(({ theme, isActive }) => css`
  padding: ${theme.spacing(16)};
  appearance: none;
  font-size: ${theme.fontSize[16]};
  margin-bottom: ${theme.spacing(16)};
  border: ${theme.borders[1]};
  background-color: ${isActive ? theme.palette.info : 'initial'};
`);

const StyledSpan = styled.span(({ theme }) => css`
  padding: ${theme.spacing(4, 8)};
  border-radius: 50%;
  margin-left: ${theme.spacing(8)};
  background: ${theme.palette.background};
`);

interface ISenderItem extends IStyledButtonProps, ComponentPropsWithRef<'button'> {
  name: ISender['name'];
  count: ISender['count'];
}

export const SenderItem = forwardRef<HTMLButtonElement, ISenderItem>(
  ({ name, count, ...rest }, ref) => (
    <StyledButton ref={ref} {...rest}>
      {name}
      <StyledSpan>{count}</StyledSpan>
    </StyledButton>
  ),
);
