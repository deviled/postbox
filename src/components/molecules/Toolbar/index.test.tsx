import React from 'react';
import { render } from '__tests__/utils';
import { Button } from 'components/atoms/Button';
import { Input } from 'components/atoms/Input';
import { Toolbar } from '.';

describe('Toolbar.tsx', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(
      <Toolbar>
        <Button>Sort</Button>
        <Input placeholder="Search..." />
      </Toolbar>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
