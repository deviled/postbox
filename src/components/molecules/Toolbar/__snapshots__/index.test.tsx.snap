// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar.tsx should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px 32px;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
  font-size: 1rem;
  border: none;
  background-color: #2196f3;
}

.c1:hover {
  -webkit-filter: brightness(85%);
  filter: brightness(85%);
}

.c1:disabled {
  -webkit-filter: initial;
  filter: initial;
  background-color: #efefef;
}

.c2 {
  font-size: 1rem;
  color: #121212;
  padding: 12px;
  border: 1px solid #cccccc;
  background-color: #ffffff;
}

.c2:hover {
  border-color: #121212;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<body>
  <div>
    <header
      class="c0"
    >
      <button
        class="c1"
      >
        Sort
      </button>
      <input
        class="c2"
        placeholder="Search..."
      />
    </header>
  </div>
</body>
`;
